### PR TITLE
qemu_guest_agent: Add steps to set selinux boolean value as 'on' 

### PR DESCRIPTION
**Pull request by 6-dehan**
Set selinux boolean value as 'On' before running some cases .

ID: 1870447
Signed-off-by: Dehan Meng \<demeng@redhat.com\>

### CODE REVIEW

**Comment by 6-dehan** [comment]
(3/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_fsfreeze.fsfreeze_list.q35: PASS (405.95 s)
(3/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_umount_frozen.q35: PASS (108.11 s)
(3/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_nonexistent_cmd.q35: PASS (111.44 s)
(3/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_fstrim.q35: PASS (119.40 s)

File: qemu/tests/qemu_guest_agent.py [hunk 1]
```diff
@@ -2057,6 +2057,20 @@ def gagent_check_fsfreeze_list(self, test, params, env):
...
+        set_selinux_bool_guest = session.cmd(set_selinux_bool_cmd).strip()
+        value_selinux_bool_guest = session.cmd_output(get_selinux_bool_cmd).strip()
+        if value_selinux_bool_guest != 'on':
+            test.error("Set selinux boolean virt_qemu_ga_read_nonsecurity_files failed.\n"
+                       "Selinux boolean virt_qemu_ga_read_nonsecurity_files is not %s."
+                       % value_selinux_bool_guest)
+        else:
+            logging.info("Selinux boolean virt_qemu_ga_read_nonsecurity_files is %s."
+                         % value_selinux_bool_guest)
+
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:2073]: Could you make a function for selinux?

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 2]
```diff
@@ -140,6 +140,8 @@
             time_service_start_cmd = net start w32time
             pause_time = 180
         - check_fstrim:
+            setsebool_cmd = "setsebool virt_qemu_ga_read_nonsecurity_files on"
```
Comment (xiagao) [inline qemu/tests/cfg/qemu_guest_agent.cfg:143]: These configs are the same in every test case,could you take them out as a common config?

**Review by xiagao — CHANGES_REQUESTED** [review]

**Comment by 6-dehan** [comment]
@xiagao CI is pass,can you have a check as well when you're free?

File: qemu/tests/qemu_guest_agent.py [hunk 3]
```diff
@@ -343,6 +343,24 @@ def gagent_verify(self, params, vm):
...
+        set_selinux_bool_guest = session.cmd(set_selinux_bool_cmd).strip()
+        value_selinux_bool_guest = session.cmd_output(get_selinux_bool_cmd).strip()
+        if value_selinux_bool_guest != 'on':
+            self.test.error("Set selinux boolean virt_qemu_ga_read_nonsecurity_files failed.\n"
+                            "Selinux boolean virt_qemu_ga_read_nonsecurity_files is not %s."
+                            % value_selinux_bool_guest)
+        else:
+            logging.info("Selinux boolean virt_qemu_ga_read_nonsecurity_files is %s."
+                         % value_selinux_bool_guest)
+        session.close()
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:362]: if you use self._open_session_list.append(session),there is no need to close.
￼

Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py:362]: And why don't you use the session already existed.

File: qemu/tests/qemu_guest_agent.py [hunk 4]
```diff
@@ -2601,6 +2620,8 @@ def gagent_check_nonexistent_cmd(self, test, params, env):
         """
         session = self._get_session(params, None)
         self._open_session_list.append(session)
+        self.gagent_setsebool_verify(self.params, self.vm)
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:2623]: I think this step is not suitable  for windows guest,right?

File: qemu/tests/qemu_guest_agent.py [hunk 5]
```diff
@@ -2719,6 +2740,7 @@ def wrap_windows_cmd(cmd):
         session = self._get_session(params, self.vm)
         self._open_session_list.append(session)
         image_size_stg0 = params["image_size_stg0"]
+        self.gagent_setsebool_verify(self.params, self.vm)
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:2743]: as the above.

File: qemu/tests/qemu_guest_agent.py [hunk 6]
```diff
@@ -3467,6 +3489,7 @@ def get_blocks():
 
         session = self._get_session(params, None)
         self._open_session_list.append(session)
+        self.gagent_setsebool_verify(self.params, self.vm)
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:3492]: as the above

File: qemu/tests/qemu_guest_agent.py (hunk 4 above)
Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py:2623]: there is no need to add self. 
 params is okay.

File: qemu/tests/qemu_guest_agent.py [hunk 7]
```diff
@@ -343,6 +343,23 @@ def gagent_verify(self, params, vm):
         self.gagent.verify_responsive()
         logging.info(self.gagent.cmd("guest-info"))
 
+    @error_context.context_aware
+    def gagent_setsebool_verify(self, params, vm):
+        session = self._get_session(params, self.vm)
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:348]: session = self._get_session(params, vm)  is okay.

File: qemu/tests/qemu_guest_agent.py [hunk 8]
```diff
@@ -343,6 +343,23 @@ def gagent_verify(self, params, vm):
...
+        self._open_session_list.append(session)
+        error_context.context("Check selinux boolean value.", logging.info)
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:350]: Set selinux xxx to xxx

File: qemu/tests/qemu_guest_agent.py [hunk 9]
```diff
@@ -343,6 +343,23 @@ def gagent_verify(self, params, vm):
...
+        set_selinux_bool_cmd = self.params['setsebool_cmd']
+        get_selinux_bool_cmd = self.params['getsebool_cmd']
+        set_selinux_bool_guest = session.cmd(set_selinux_bool_cmd).strip()
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:353]: Can get value of  set_selinux_bool_guest to run cmd() ?

File: qemu/tests/qemu_guest_agent.py [hunk 10]
```diff
@@ -343,6 +343,23 @@ def gagent_verify(self, params, vm):
...
+        value_selinux_bool_guest = session.cmd_output(get_selinux_bool_cmd).strip()
+        if value_selinux_bool_guest != 'on':
+            self.test.error("Set selinux boolean virt_qemu_ga_read_nonsecurity_files failed.\n"
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:356]: test.error() is okay.

File: qemu/tests/qemu_guest_agent.py [hunk 11]
```diff
@@ -343,6 +343,23 @@ def gagent_verify(self, params, vm):
         self.gagent.verify_responsive()
         logging.info(self.gagent.cmd("guest-info"))
 
+    @error_context.context_aware
+    def gagent_setsebool_verify(self, params, vm):
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:347]: if it's better to use function name of  gagent_setsebool_on()?
And can you add docstring for this function?

File: qemu/tests/qemu_guest_agent.py (hunk 10 above)
Comment (6-dehan) [reply qemu/tests/qemu_guest_agent.py]: The 'self' looks like to add to.
E0602:356,12: QemuGuestAgentTest.gagent_setsebool_on: Undefined variable 'test'

File: qemu/tests/qemu_guest_agent.py [hunk 12]
```diff
@@ -343,6 +343,30 @@ def gagent_verify(self, params, vm):
...
+                              "as on.", logging.info)
+        set_selinux_bool_cmd = self.params['setsebool_cmd']
+        get_selinux_bool_cmd = self.params['getsebool_cmd']
+        session.cmd(set_selinux_bool_cmd).strip()
+        value_selinux_bool_guest = session.cmd_output(get_selinux_bool_cmd).strip()
+        if value_selinux_bool_guest != 'on':
+            self.test.error("Set selinux boolean virt_qemu_ga_read_nonsecurity_files failed.\n"
+                            "Selinux boolean virt_qemu_ga_read_nonsecurity_files is not %s."
+                            % value_selinux_bool_guest)
+        else:
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:366]: I think you can delete the else as you have mentioned what you are doing.

File: qemu/tests/qemu_guest_agent.py [hunk 13]
```diff
@@ -3467,6 +3495,8 @@ def get_blocks():
 
         session = self._get_session(params, None)
         self._open_session_list.append(session)
+        if params.get("os_type") == "linux":
+            self.gagent_setsebool_on(params, self.vm)
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:3499]: And as you set selinux config, could you change them back when you finish your test?

**Review by xiagao — CHANGES_REQUESTED** [review]

**Comment by 6-dehan** [comment]
Need to create new repo for these patches.

**Pull request closed without merging**